src/app/transactions/fields/ValidatorFormField.tsx

```diff
@@ -21,6 +22,12 @@ export function ValidatorFormField({
   validators,
   setDecimals,
 }: ValidatorFormFieldProps) {
+  // Watch the chainId to trigger re-render
+  const chainId = form.watch("chainId");
+
+  // Log chainId and validatorIndex to track changes
+  useEffect(() => {}, [chainId, form]);
```

Comment:
I'm not sure what this does 🤔

Reply:
The goal of this is to:
- monitor which asset (chainId) is selected by the user in the staking flow (cosmos, dydx, etc...)
- if the chainId changes (meaning if the user first selected an ATOM asset, but then change his mind and select dYdX for instance), then the form is reset to avoid the old list of validators (cosmos validators) to be displayed instead of the new one (dydx validator).

The useEffect() can be safely removed I think.

Reply:
https://github.com/AdamikHQ/adamik-app/pull/57/commits/6fcf34e78927c23d7a3c7bfabd373b55a5920249